### DISCUSSION TITLE
Ability to hook into adding custom fields to checkout

### DISCUSSION THREAD

**Discussion by nickfmc** [woocommerce/woocommerce-gutenberg-products-block#5736] [Q&A - Extension Integrations and Customizations] [answered]
We really need a way to add custom fields to the checkout block. I think this is one big thing holding it back

**Comment by alexflorisca** [comment, marked as answer]
Hi @nickfmc, thanks for raising the issue. You can currently add custom fields to the checkout by adding a new inner block to the address area, and render an input inside of it. We do something similar in [this test plugin](https://github.com/woocommerce/newsletter-test) to render a checkout box for subscribing to a newsletter. You can follow the same format to add custom fields elsewhere on the Checkout block.

**Comment by webdados** [comment]
We all know blocks and react are the brand new world of WordPress and WooCommerce, but a lot of developers will desist from developing for WooCommerce if they're not given tools to easily customize, for example, the checkout.

Adding a custom field to the regular checkout is as easy as using a filter like `woocommerce_billing_fields` or `woocommerce_checkout_fields`. As we can see by the test plugin, adding a custom field to the block-based checkout is an all-other level of complexity that is just too much for what should be such a simple task.

There should be an easier way for developers to register simple custom checkout fields and have WooCommerce render them without having to touch on React of even explicitly having to go to the block editor and add that field manually.

**Comment by opr** [comment]
Hi - closing this discussion in favour of https://github.com/woocommerce/woocommerce-blocks/discussions/11173 where we will discuss the feature as we work on it.  It'll be better to focus all communication in a single discussion, rather than the few we have open right now for this.

Thanks!
